### DISCUSSION TITLE
2.7.1 正式版更新错误的情况说明

### DISCUSSION THREAD

**Discussion by LTCatt** [Meloong-Git/PCL#3651] [讨论]
### 情况说明：
由于正式版 build 脚本只会选择 master 分支，而我另开了一个分支做 2.7.1 的临时修复，master 上是 `开发中的 2.7.2`，导致 4/8 凌晨 1 点更新的 `正式版 2.7.1` 错误地编译、上传成了 `开发中的 2.7.2`。现已经于 4/8 17:15 重新上传了正确的 `正式版 2.7.1`，快照版未受影响。
使用错误版本（版本号错误显示为 2.7.0）的正式版用户现已可以正常更新。**该错误版本存在严重问题，请务必进行更新！**
如果使用该错误版本安装了 Forge/Fabric，**请在更新后重新安装该版本或整合包**。该错误版本安装的 Forge/Fabric 已确认无法加载 Mod（#3646）。
如果使用该错误版本启动时出现内存不足崩溃，在更新后即可恢复正常。
由于这是临时紧急修复，病毒误报反馈还在处理中，被 Windows Defender 报毒是正常情况，加白名单即可。
已经对该问题添加了检查项，之后该问题不会再次出现。非常抱歉！感谢大家的理解 Orz！

### 关于 `2.7.1.1` 更新公告：
为了避免有人觉得 “我已经更新了 2.7.1 了为啥还要更新”，重新上传版本在正式版更新说明中描述为 `2.7.1.1` 且附带了更新日志。
**这只是为了便于普通用户理解**，让不关注技术细节的人也能知道这是一个新的修复后的版本，实际上版本并没有变，依然是 `2.7.1`。

### 相关 issue：
该版本尚未完成开发，所以没有修改版本号，导致了 #3647、#3648 。
收到多个独立来源的 Out Of Memory 反馈，它们应当是原本预计于 `快照版 2.7.2` 测试的 #3511 导致，该改动将在正式发布前回滚。
#3646 由原本预计于 `快照版 2.7.2` 测试的 #2801 还没有完全做完导致。

### 版本鉴别：
为便于社区鉴别错误版本与正常版本，说明一下怎么看该版本有没有问题。
版本号 `2.7.1` 的是正常版本，版本号 `Snapshot 2.7.0`（快照版 2.7.0）的是正常版本。
版本号 `Release 2.7.0`（正式版 2.7.0）的，如果日志开头有 “系统编码” 一行则为错误版本，没有就是正常的 2.7.0 老版本。
![image](https://github.com/Hex-Dragon/PCL2/assets/40348278/2bee883e-bb2a-493b-9ae4-c019ff43cc00)

**Comment by MoYuan-CN** [comment]
啊这……
~~年度更新事故~~

**Reply by ImLee11** [reply]
确实

**Reply by wuliaodexiaoluo** [reply]
我就说怎么安装好 Fabric 但 Minecraft 根本识别不到。。。。
这个版本是不是能卡进开发者模式。。。。

**Reply by LTCatt** [reply]
不能，错误上传的是没做完的正式版 2.7.2，但还是正式版

**Reply by MoYuan-CN** [reply]
构建的时候会自动给那些开发者内容禁用掉的

**Reply by wuliaodexiaoluo** [reply]
可惜了，还想体验下开发者模式

**Comment by MoYuan-CN** [comment]
@LTCatt 打错反馈号了喂（#2801 导致的是 #3646，46 46 46！）

**Reply by wuliaodexiaoluo** [reply]
az

**Reply by LTCatt** [reply]
很急，改了（

**Comment by wuliaodexiaoluo** [comment]
上面那个还是 3606
>如果使用该错误版本安装了 Fabric，请在更新后重新安装 Fabric。该错误版本安装的 Fabric 已确认无法加载 Mod #3606
如果使用该错误版本启动时出现内存不足崩溃，在更新后即可恢复正常。

**Comment by wuliaodexiaoluo** [comment]
> ### 版本号鉴别：
> 为便于社区鉴别错误版本与正常版本，说明一下怎么看该版本有没有问题。 版本号显示 `2.7.1` 的是正常版本，版本号显示为 `Snapshot 2.7.0`（快照版 2.7.0）的是正常版本。 版本号显示为 `Release 2.7.0`（正式版 2.7.0）的，如果日志开头有 “系统编码” 一行则为错误版本，没有就是正常的 2.7.0 老版本。 ![image](https://private-user-images.githubusercontent.com/40348278/320437820-2bee883e-bb2a-493b-9ae4-c019ff43cc00.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI1NzA3MTksIm5iZiI6MTcxMjU3MDQxOSwicGF0aCI6Ii80MDM0ODI3OC8zMjA0Mzc4MjAtMmJlZTg4M2UtYmIyYS00OTNiLTlhZTQtYzAxOWZmNDNjYzAwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDA4VDEwMDAxOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTNiMTNhNzVhMzAzMzcwMzY0YjM4MDRmNzM2OGQzMzk0OTc2MGQxMWRmNGRkOTcwMzZhNTgxZDFmOWE2NGQ0MDAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.p8D0bkJiO_9K3ujyOeo9ixIF52ZbYBgIhOvBddarTk4)

事实上，可以直接看的，错误版本的数字是 319.50

**Reply by LTCatt** [reply]
正常的 2.7.0 正式版也是 319.50

**Reply by wuliaodexiaoluo** [reply]
az
6

**Reply by wuliaodexiaoluo** [reply]
不对啊 Release 2.7.0 不是 319 吗？

**Reply by MoYuan-CN** [reply]
SnapShot 是整数，Release 是 .50
~~系统编码是不是给开发者工具包那个vb打包进去了（笑~~

**Comment by LTCatt** [comment]
情况说明已更新完毕，追加了一些额外的信息，如果之前看过的可以再看一次……？

**Comment by ghost** [comment]
打卡年度事故（保留节目1/1

**Comment by iPodtouch6557** [comment]
借楼一说，我觉得
其实最好区分快照和公测版本号，格式完全一样我自己也搞不清……

**Comment by wuliaodexiaoluo** [comment]
>> SnapShot 是整数，Release 是 .50
系统编码是不是给开发者工具包那个vb打包进去了（笑
>
```
[19:34:47.349] [Start] 程序版本：Snapshot 2.7.1（320）
[19:08:20.197] [Start] 程序版本：Release 2.7.1（321）
```
So？

**Reply by wuliaodexiaoluo** [reply]
![image](https://github.com/Hex-Dragon/PCL2/assets/160083863/ce884827-8b38-4139-a6c2-213856bd00a1)

关键他真就写的 321.50

**Reply by wuliaodexiaoluo** [reply]
但我寻思着319不是2.7.0吗？
